docs/src/pages/components/data-grid/filtering/CustomRatingFilterOperator.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import { Rating } from '@material-ui/lab';
+import {
+  PreferencePanelsValue,
+  DataGrid,
+  FilterInputValueProps,
+  NUMERIC_OPERATORS,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+
+const useStyles = makeStyles({
+  ratingContainer: {
+    display: 'inline-flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    height: 48,
+    paddingLeft: 20,
+  },
+});
+
+function RatingInputValue(props: FilterInputValueProps) {
+  const classes = useStyles();
+  const { item, applyValue } = props;
+  const [filterValueState, setFilterValueState] = React.useState<number>(
+    Number(item.value),
+  );
```

Comment:
The component is controlled. I don't understand why we have a state

```suggestion
  const filterValueState = Number(item.value);
```

Reply:
The value can change by using the prop filterModel and also using the UI so we control the value

Reply:
Ok, so you are confirming it? The InputComponent should be stateless. It would be great to mention it in the documentation.

Reply:
Yes correct. Cleaning up ;)